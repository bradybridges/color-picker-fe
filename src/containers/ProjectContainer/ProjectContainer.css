.project-container {
  display: grid;
  /* grid-gap: 20px 15px; */
  /* padding: 20px 20px; */
  grid-template-columns: repeat(auto-fit, minmax(300px, 0.5fr));
  
  justify-content: center;
  overflow: scroll;
}