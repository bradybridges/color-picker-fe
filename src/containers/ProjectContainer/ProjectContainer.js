import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './ProjectContainer.css';
import ProjectCard from '../ProjectCard/ProjectCard';


export const ProjectContainer = ({ projects }) => {
  const displayProjectCards = projects.map((project, index) => {
    return (
      <ProjectCard
        {...project}
        key={index}
        className='project'
      />
    );
  });

  return (
    <section className='project-container'>
      {displayProjectCards}
    </section>
  );
};

export const mapState = (state) => ({
  projects: state.projects,
});

export default connect(mapState)(ProjectContainer);

ProjectContainer.propTypes = {
  projects: PropTypes.array.isRequired,
};
