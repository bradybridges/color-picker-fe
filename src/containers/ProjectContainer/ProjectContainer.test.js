import { ProjectContainer, mapState } from './ProjectContainer'
import { shallow } from 'enzyme'
import React from 'react'

describe('ProjectContainer', () => {
  describe('ProjectContainer component', () => {
    let wrapper;
    beforeEach(() => {
      const mockProjects = [
        {
          id: 1,
          name: 'a project'
        }, 
        {
          id: 2,
          name: 'another project'
        }
      ]

      wrapper = shallow(<ProjectContainer projects={mockProjects}/>)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapState', () => {
    it('should return an object with the projects array', () => {
      const mockState = {
        projects: [
          {
            id: 1,
            name: 'a project'
          }, 
          {
            id: 2,
            name: 'another project'
          }
        ],
        palettes: [
          {
            id: 1,
            project_id: 1,
            name: 'a palette',
            color_1: '#FFFFFF',
            color_2: '#FFFFFF',
            color_3: '#FFFFFF',
            color_4: '#FFFFFF',
            color_5: '#FFFFFF',
          }
        ]
      }
        const expectedState = {
          projects: [
            {
              id: 1,
              name: 'a project'
            }, 
            {
              id: 2,
              name: 'another project'
            }
          ]
        }

        const mappedProps = mapState(mockState)

        expect(mappedProps).toEqual(expectedState)
      })
    })
  })
