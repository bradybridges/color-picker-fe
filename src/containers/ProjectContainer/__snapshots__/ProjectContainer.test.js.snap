// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer ProjectContainer component should match the snapshot 1`] = `ShallowWrapper {}`;
